Source/CompSpinalEngineTrail.cs

```diff
@@ -0,0 +1,179 @@
+﻿using RimWorld;
+using SaveOurShip2;
+using System.Text;
+using Verse;
+
+namespace TheCafFiend
+{
+    [StaticConstructorOnStartup]
+    public class CompSpinalEngineTrail : CompEngineTrail
+    {
+        public override string CompInspectStringExtra()
+        {
+            StringBuilder stringBuilder = new StringBuilder();
+            
+            if (fullyFormed)
+            {
+                stringBuilder.Append($"Fuel burn rate per second: {cachedFuelUse} of {cachedFuelAllowed} supported");
+                stringBuilder.AppendInNewLine($"Thrust: {cachedThrust * 500}");
+            }
+            else
+            {
+                stringBuilder.Append(cachedError.Colorize(UnityEngine.Color.red));
+            }
+            return stringBuilder.ToString();
+        }
+        public virtual CompProperties_SpinalEngineTrail Properties //Wait is this dirty I just want to be clear it's not hiding hrm
+        {
+            get { return props as CompProperties_SpinalEngineTrail; }
+        }
+        public class ReturnValue
+        {
+            public float thrustAmp;
+            public float fuelAmp;
+            public float fuelAllowAmp;
+            public float powerUseAmp;
+            public int supportWeight;
+            public string playerError;
+            public bool fullyFormed;
+        }
+        private int cachedThrust = 0;
+        private int cachedMass = 0; // add/remove on the *engine* does this itself, this is only amps+caps
+        public bool fullyFormed = false;
+        private int cachedFuelUse = 0;
+        private int cachedPowerUse = 0;
+        private int cachedFuelAllowed = 0; // Error below probably never surfaces to a player but I want a fallback
+        private string cachedError = "Spinal engine isn't fully formed! Check for engine accelerators, and all three components of the fuel infrastructure!";
+        public override int Thrust
+        {
+            get
+            {
+                if (parent.TryGetComp<CompSpinalEngineTrail>() == null) //should never? happen but reflex
+                {
+                    Log.Message("SOS2 spinal engines: somehow hit thrust from a non-compspinalenginetrail?");
+                    return base.Thrust;
+                }
+                if (fullyFormed)
```

Comment:
Wouldn't that leave it with no return value at all if it was fully formed? Even compressing the if, which normally I'm a little reluctant to do but this is hecken short, wouldn't it be more like the below? Or does this single line if only evaluate the first ..... chuck delimited by a ;, statement right? 
if (!fullyformed) calc();
return cachedThrust;

Reply:
Oh, yeah sorry github broke my new line formatting.

if (!fullyFormed)
  calc();

return cachedThrust;

Technically newlines are only there for our convenience.  The compile will strip them out in the parsing stage because they're worthless in C-style languages.  Python on the other hand will give you shit about spaces and tabs and newlines all day long.

Reply:
I should add - semicolons denote the end of a statement, so you can do things like have multiple statements on one line if they make sense:
x = oldX*2; y = oldY*3; z = oldZ;

Since I know you like to know more AND you like rules, you might be interested in [Formal Grammar](https://en.wikipedia.org/wiki/Formal_grammar), which is one way we think about and design coding languages.  You may also be interested in [C#'s language spec](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/grammar).

Reply:
Halp I lost time wanting to know more that is very comprehensive dang